promgen/migrations/0031_migrate_parent_object_for_audit_log.py

```diff
@@ -0,0 +1,39 @@
+# Generated by Django 4.2.11 on 2025-12-08 15:30
+import json
+
+from django.db import migrations
+
+from promgen import models
+
+
+def extract_data_and_populate_parent_object(apps, schema_editor):
+    MODELS = {
+        "project": models.ContentType.objects.get(app_label="promgen", model="project"),
+        "service": models.ContentType.objects.get(app_label="promgen", model="service"),
+        "farm": models.ContentType.objects.get(app_label="promgen", model="farm"),
+    }
+
+    for log in models.Audit.objects.all():
+        if log.parent_content_type_id and log.parent_object_id:
+            continue
+        data = json.loads(log.data)
+
+        if "content_type" in data and "object_id" in data:
+            log.parent_content_type_id = data["content_type"]
+            log.parent_object_id = data["object_id"]
+            log.save()
+        else:
+            for key in MODELS:
+                if key in data:
+                    log.parent_content_type_id = MODELS.get(key).id
+                    log.parent_object_id = data[key]
+                    log.save()
+                    break
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("promgen", "0030_audit_parent_content_type_id_audit_parent_object_id_and_more"),
+    ]
+
+    operations = [migrations.RunPython(extract_data_and_populate_parent_object)]
```

Comment:
If we populate the columns in the other migration file, this one becomes unnecessary.
Please remove it: 
[0003-Revert-Add-a-migration-script-for-migrate-old-audit-.PATCH](https://github.com/user-attachments/files/21747736/0003-Revert-Add-a-migration-script-for-migrate-old-audit-.PATCH)